utils/urlUtils.js

```diff
@@ -0,0 +1,26 @@
+function getUrlDetails(url) {
+    let urlDetails = {
+        project : "",
+        type : "",
+        id : ""
+    }
+
+    if(url.lastIndexOf('pull') === -1)
+        urlDetails.type = "issue";
+    else if(url.lastIndexOf('issues') > url.lastIndexOf('pull'))
+        urlDetails.type = "issue";
+    else
+        urlDetails.type = "pull";
+
+    let idPart = url.slice(url.lastIndexOf(urlDetails.type)+(urlDetails.type).length,url.length)
+    urlDetails.id = idPart.match(/\d+/)[0];
+
+    url = url.slice(0,url.lastIndexOf(urlDetails.type));
+    url = url.slice(url.indexOf("github.com/coding-blocks")+24,url.length);
+
+    urlDetails.project = url.replace(new RegExp('/', 'g'), '');
```

Comment:
pretty much unreadable.

Reply:
what can I change? the logic is clear. when I got the type(pull or issue),  id is after that. I used a number regex at that part. for getting project name I used string be github.com/codingblocks and link type